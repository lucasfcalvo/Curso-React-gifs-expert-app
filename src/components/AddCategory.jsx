import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewValue }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    // setCategories( categories => [ inputValue, ...categories ]);
    setInputValue("");
    onNewValue(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
