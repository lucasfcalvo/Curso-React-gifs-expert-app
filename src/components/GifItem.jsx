export const GifItem = ({title, url,}) => {
    // console.log(image)
  return <div className="card"> 
  <img src={url} alt="img" />
  <p>{title}</p>

    
  </div>;
};
