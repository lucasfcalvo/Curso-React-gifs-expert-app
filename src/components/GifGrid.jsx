import { GifItem } from "../components";
import PropTypes from "prop-types";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  console.log({ images, isLoading });
  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h6>Cargando...</h6>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
