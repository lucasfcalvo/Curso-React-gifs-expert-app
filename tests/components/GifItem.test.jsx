import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba de componente <GifItem/>", () => {
  const title = "saitama";
  const url = "https://one-punch.saitama/soy";
  test("Debe de hacer match con el snapshot ", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("Debe de mostrar la imagen con el URL y el Titulo indicado", () => {
    // 
    render(<GifItem title={title} url={url} />);
    expect(screen.getByRole("img").src).toBe(url);
    expect(screen.getByRole("img").alt).toBe(title);
  });
});
