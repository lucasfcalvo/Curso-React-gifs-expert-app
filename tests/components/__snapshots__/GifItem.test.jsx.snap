// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente <GifItem/> Debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-punch.saitama/soy"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
